### PR TITLE
[Site Editor]: Fix ability to edit trashed pages

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Thanks to @aaronrobertshaw 's [comment](https://github.com/WordPress/gutenberg/pull/60230#discussion_r1540650574) I realised that we allow pages with `trash` status to be edited in site editor. This shouldn't be the case and this PR fixes that.

## Testing Instructions
1. In pages list apply the `trash` status filter. Observe that there is no link to the title.
2. In `list` layout observe that the preview is not redirecting to edit the page.

### CODE REVIEW

**Comment by Mamaduka** [comment]
@ntsekouras, it looks like I can still use <kbd>Enter</kbd> or <kbd>Space</kbd> to enter the edit mode.

Should the component aslo indicate that editing of the page canvas is disabled for a11y?

## Screencast

https://github.com/WordPress/gutenberg/assets/240569/73cb27bf-0fc7-4ee9-9c9b-6ff9ad3b7653

**Comment by ntsekouras** [comment]
>it looks like I can still use Enter or Space to enter the edit mode.
>Should the component aslo indicate that editing of the page canvas is disabled for a11y?

Thanks for catching this! I updated.

Regarding the `a11y`, I think adding `aria-disabled` can be enough for this case.. --cc @afercia

File: packages/edit-site/src/components/block-editor/editor-canvas.js
Comment (Mamaduka) [inline packages/edit-site/src/components/block-editor/editor-canvas.js:89]: Maybe we should disable all interaction callbacks in a single place and short-circuit `onClick` here (line 94) instead of the router?

https://github.com/WordPress/gutenberg/blob/c7036a5a70ce88d4d9014a6ccbd45cd131f9852d/packages/edit-site/src/components/block-editor/editor-canvas.js#L78-L84

Comment (ntsekouras) [reply packages/edit-site/src/components/block-editor/editor-canvas.js:89]: Yeah, this is probably better.

**Review by Mamaduka — APPROVED** [review]
Thanks for the followups, Nik! The changes test well for me ✅

Let's wait for the a11y feedback before landing this.

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for fix @ntsekouras 👍 

This is testing well for the dataviews `list` layout but it's still possible to edit a trashed page via the `grid` layout. 

The `grid` layout's buttons are [passed](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L185) an `onClick` handler directly [from `useLink`](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L158-L162).


https://github.com/WordPress/gutenberg/assets/60436221/e7f61065-c85d-4369-b704-0d9346c7e04c

**Comment by ntsekouras** [comment]
>but it's still possible to edit a trashed page via the grid layout.

Thanks for catching this! I updated to handle featured image too.

**Review by aaronrobertshaw — APPROVED** [review]
Appreciate the quick turnaround on the feedback @ntsekouras 🚀 

After pulling the latest, I can no longer edited a trashed page via the `grid` view. Once this gets a green light from the a11y perspective, it should be good to go.

As a follow-up, it might be good if we could prevent some of the block selection behaviour in the page preview for the trashed pages. 


https://github.com/WordPress/gutenberg/assets/60436221/5e10717a-12a8-4e05-9880-aff5d5c5f293

**Comment by afercia** [comment]
From an a11y perspective, as long as the controls to switch to edit mode:
- are noop'ed in some way
- have an aria-disabled attribute

That would be sufficient.
However, a disabled control should also _look disabled_. This is not the case, for example, in the grid view where a disabled 'card' still looks enabled ans also shows a ppointer cursor when hovered.

File: packages/edit-site/src/components/block-editor/editor-canvas.js
Comment (ntsekouras) [inline packages/edit-site/src/components/block-editor/editor-canvas.js:101]: @mcsf if I remember correctly you've looked at this event recently. Do you think it's fine to use?

**Comment by ntsekouras** [comment]
>However, a disabled control should also look disabled. This is not the case, for example, in the grid view where a disabled 'card' still looks enabled ans also shows a ppointer cursor when hovered.

@afercia I updated not to show a cursor pointer when disabled. It feels a bit weird to have disabled styles in featured image though if it has no action. Maybe the solution is just not wrap in a button when disabled or it's enough as is?

**Comment by afercia** [comment]
> @afercia I updated not to show a cursor pointer when disabled. It feels a bit weird to have disabled styles in featured image though if it has no action. Maybe the solution is just not wrap in a button when disabled or it's enough as is?

@ntsekouras thanks for the ping, not sure I fuly follow though. My point is that if we disable interaction with an UI control, the UI should also visually indicate that the control is disabled. Instead, especially in the Grid view, there's no difference betwen a control (the card) that is enabled and one that is disabled. Screenshot of the same page card when published and when in the trash:

Screenshot 2024-04-03 at 10 38 42

File: packages/edit-site/src/components/block-editor/editor-canvas.js
Comment (mcsf) [reply packages/edit-site/src/components/block-editor/editor-canvas.js:101]: Yeah, this might be fine.

Indeed, I proposed using a capture-phase listener in #59563 to allow clicking on a whole row to select it. It's worth noting (and I had forgotten about this in our recent chat) that I ended up removing it in #59803 because the requirements had changed. But if the goal is to disable interactions going into the canvas, this may be good.

I expressed some of my concerns with `on*Capture` as follows:

> Mostly I think it’s important to remain aware that this is basically a lateral system. Technically it’s part of the same event propagation system, but since we never use Capture, it could lead to blind spots in our mental model, might make things trickier to debug… I’m saying all this in abstract, as potential risks, because I was the one suggesting Capture to begin with :)
>
> For instance when we debug a problem we may inspect the node from an element down to its leaves to find some problematic listener, and we forget to inspect in the other direction: looking for problems near the document root

In this particular case I'd also pay attention to the fact that we want to make the canvas "mostly" inert but not completely. What does this mean for event interception? Plus, do users and devices understand that this area is now partially disabled? Etc.

**Comment by ntsekouras** [comment]
>My point is that if we disable interaction with an UI control, the UI should also visually indicate that the control is disabled.

I updated the code to **not** render a button at all when the featured image is not actionable, so it should be fine a11y wise as we don't render a disabled control.

@jameskoster I'd love a sanity check on the css used  for the featured image in pages list whether I missed any small detail. In my testing looks identical to trunk, but designer's eyes are better trained than mine :)

**Comment by jameskoster** [comment]
Thanks for the ping, looks good to me. 

Related; we should probably adjust the primary field color to differentiate when it's a link or not, it's a bit misleading at the moment. I'll push a tweak.

**Comment by jameskoster** [comment]
The appearance of the frame is a bit strange on-click in the Trash view. The focus ring appears which kind of suggests something can/is happening. 

Is the focus ring required for a11y? If not, I'd suggest hiding it when the frame is non-interactive. If it is required, can it be applied on `focus-visible` instead to avoid showing it on click? --cc @afercia

File: packages/edit-site/src/components/block-editor/editor-canvas.js
Comment (ntsekouras) [reply packages/edit-site/src/components/block-editor/editor-canvas.js:101]: >do users and devices understand that this area is now partially disabled? Etc.

I think so yes, since we have aria-disabled and role='button'

**Comment by afercia** [comment]
> Is the focus ring required for a11y? If not, 

Worth noting other blue outlines may appear when clicking on the arira-disabled iframe of a trashed page, for example the outline around a 'selected' block. Screenshot:

Screenshot 2024-04-04 at 14 07 30

> I'd suggest hiding it when the frame is non-interactive.

It is true that the iframe (semantically it's a button) is non-interactive but it is still focusable. As such, it should show a focus style when focused. Not sure this is a native focus style that can use `focus-visible` but if it's technically possible I woulnd't have objections to not show the focus style while the iframe is clicked, as long as the focus style is still visible when the iframe is focused.

**Comment by jameskoster** [comment]
Thanks for clarifying. If it's technically possible I think that would be a nice change to include.

**Comment by ntsekouras** [comment]
>Worth noting other blue outlines may appear when clicking on the arira-disabled iframe of a trashed page, for example the outline around a 'selected' block

I have updated the code and the click event doesn't propagate and apply focus styles to blocks, etc..

So the only focus style that is visible is at the iframe(semantically is a button), when focused with keyboard or a click. I think that's expected and wanted. 

Is anything else to address in this PR or we can ship?

**Comment by jameskoster** [comment]
Did you push? I'm still seeing the outline on click.

**Comment by ntsekouras** [comment]
>Did you push? I'm still seeing the outline on click.

>So the only focus style that is visible is at the iframe(semantically is a button), when focused with keyboard or a **click**. I think that's expected and wanted.

I think it's fine to have the outline because when clicking the iframe it becomes focused, so I think we shouldn't remove the outline.

**Comment by jameskoster** [comment]
I don't think it makes sense to show the focus styles on click since the button is disabled – it suggests there is some possible interaction which in this case there is not?

**Comment by ntsekouras** [comment]
>I don't think it makes sense to show the focus styles on click since the button is disabled – it suggests there is some possible interaction which in this case there is not?

We do that for other buttons too for a11y, for example in disabled save button..

**Comment by jameskoster** [comment]
The save button is slightly different – it needs to remain focussed when you click Save so that you don't experience focus loss. I think this case is slightly different; once you select a page in trash, the preview button is instantly disabled. I understand it's helpful to show focus styles when you tab to it, but I'm not really sure what benefit is to showing the focus styles on click? If anything we should inform mouse users that the button is disabled some how.

**Comment by ntsekouras** [comment]
>The save button is slightly different

Yeah, you're right about that.

>Not sure this is a native focus style that can use focus-visible but if it's technically possible

It's not native (so no css solution) and I've tried a few things with no success yet. This is because the `onFocus` event is fired before the `onClickCapture`. I'll try a bit more..

**Comment by jameskoster** [comment]
@ntsekouras if it's technically impractical then it's fine to leave as is. We could potentially follow up with a hover treatment that helps mouse users understand the button is disabled.

**Review by jameskoster — APPROVED** [review]

**Pull request merged**
